Switch to pipelines version of python 3.7

dmcilvaney opened the pull request:
Pipelines offers a tool to install different versions of python, should be quicker than doing a fresh build for i.Mx8.

dmcilvaney committed:
Switch to pipelines version of python 3.7

neilsh-msft commented on ci/azure-pipelines.yml:
```diff
@@ -20,23 +20,13 @@ jobs:
-
-        sudo python3.7 -m pip install --upgrade pip
-        sudo python3.7 -m pip install --upgrade setuptools
-        sudo python3.7 -m pip install mu_environment
```
Are pip and setuptools upgraded as part of the UsePythonVersion task?  Is mu_environment installed also?

dmcilvaney replied:
No, will need to fix that.
Its frustrating there isn't a way to test the pipeline without pushing commits.

christopherco replied:
Should we stand up a dev pipeline that is manually queued?

dmcilvaney replied:
Still doesn't matter, it can only run against commits that are pushed into the upstream repo.

Closest we can do it create a testing branch here we can push stuff into, but we would still need to do pull requests into that branch.

christopherco replied:
I was referring to the latter. You can make changes to a testing branch, commit, and manually queue the pipeline to test it.

dmcilvaney replied:
Sure, I can do that in the future.

dmcilvaney committed:
Restore mu python packages

dmcilvaney committed:
Update azure-pipelines.yml

python3.7 isn't found, checking version of python installed. (hopefully python3 will have been set correctly in the path).

dmcilvaney committed:
Attempting MU build fix

dmcilvaney committed:
Update azure-pipelines.yml
